data_deletion/delivered_data.py

```diff
@@ -114,6 +114,12 @@ def delete_data(self):
             self.delete_dir(self.deletion_dir)
 
         # Data has been deleted, so now clean up empty released directories
-        for folder in set(os.path.dirname(s.released_data_folder) for s in deletable_samples if s.released_data_folder):
-            if os.path.isdir(folder) and not os.listdir(folder):
-                self._execute('rm -r ' + folder)
+        for released_sample in set(s.released_data_folder for s in deletable_samples):
+            if not released_sample:
+                continue
```

Comment:
@mwhamgenomics I don't really understand these three lines of code?

Reply:
Suppose `deletable_samples` contains samples with the following values for `released_data_folder`:
```
a_project/2019-01-11/sample_3
a_project/2019-01-11/sample_2
a_project/2019-01-10/sample_1
```

Previously, we were iterating through the dirname of each released folder (e.g. `a_project/2019-01-11'), casting to a set because multiple samples can be released on the same day, and using `if s.released_data_folder` because `released_data_folder` returns `None` if the dir doesn't exist, e.g. if the sample has already been deleted.

Now we are iterating through each released folder directly, `continue`-ing if the dir doesn't exist (i.e. stop there but carry on with the rest of the loop). The `set` cast was no longer needed, so has been simplified.